extensions/2.0/Khronos/KHR_materials_volume/README.md

```diff
@@ -0,0 +1,268 @@
+</figure>
+
+The volume extension has to be combined with `KHR_materials_transmission` or `KHR_materials_translucency`. Light that falls onto the volume boundary may enter the volume, depending on the transmission or translucency of the surface's BSDF. Inside the volume, light might be absorbed or scattered by particles in the medium, and eventually hit some surface from inside the volume. It can be a different surface and will most probably be a different point in space compared to the entrance point. At this exit point, if the material properties allow, light may leave the volume.
+
+<figure style="text-align:center">
+<img src="./figures/volume.svg"/>
+<figcaption><em>Interaction of light rays inside the volume and at the boundaries. The volume is homogeneous and has an index of refraction of 1.5.</figcaption>
+</em></figure>
+
+When light interacts with the surface, it is reflected or refracted at microfacets, taking the roughness of the material into account. The index of refraction is taken from `KHR_materials_ior`.
+
+## Extending Materials
+
+The volumetric properties are defined by adding the `KHR_materials_volume` extension to any glTF material. A non-zero thickness switches from thin-walled to volumetric behavior. This requires a manifold/closed mesh. The properties of the medium are not texturable, it is assumed to be homogeneous.
+
+```json
+materials: [
+    {
+        "extensions": {
+            "KHR_materials_volume": {
+                "thicknessFactor": 1.0,
+                "attenuationDistance":  0.006,
+                "attenuationColor": [ 0.5, 0.5, 0.5 ],
+                "subsurfaceColor": [ 0.572, 0.227, 0.075 ]
+            }
+        }
+    }
+]
+```
+
+## Properties
+
+The extension defines the following parameters to describe the volume.
+
+| | Type | Description | Required |
+|-|------|-------------|----------|
+| **thicknessFactor** | `number` | The thickness of the volume beneath the surface in meters in the local coordinate system of the node. If the value is 0 the material is thin-walled. Otherwise the material is a volume boundary. The `doubleSided` property has no effect on volume boundaries. | No, default: `0`. |
+| **thicknessTexture** | `textureInfo` | A texture that defines the thickness, stored in the G channel. This will be multiplied by `thicknessFactor`. | No |
+| **attenuationDistance** | `number` | Density of the medium given as the average distance in meters that light travels in the medium before interacting with a particle. | No, default: +Infinity |
+| **attenuationColor** | `number[3]` | The color that white light turns into due to absorption when reaching the attenuation distance. | No, default: `[1, 1, 1]` |
```

Comment:
Is this parameter here to model the effect on absorption at different wavelengths, ie similar to what happens when light is absorbed in water and blue/violet light reaches further - making a sea appear blue-ish?

If so - maybe this could be mentioned?
If not, could you please explain the physicality of this parameter?

Reply:
Yes, exactly, attenuation color is the wavelength-dependent part of the attenuation coefficient. The description in the table here is just a small teaser. The detailed relationship between attenuation color and distance is explained in section "Absorption and Subsurface Scattering" (which describes the relationshop between the parameters and the physical quantities) and, to some extend, also in section "Base Color and Absorption".

Reply:
Thanks @proog128 
While I read the `Absorption and Subsurface Scattering` section I was not able to deduce the source of the parameter.

Reply:
I will try to improve the connection between the section and the table. Something like (additions in bold):

> To provide convenient parameterization for users this extension **provides three user-friendly parameters** attenuation color a, attenuation distance d and subsurface color ρms (see section "Properties"). **We will now describe how to derive the physical parameters for absorption and scattering in the medium from the user-friendly parameters.**